llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1670,6 +1672,8 @@ void GCNPassConfig::addPreEmitPass() {
   // cases.
   addPass(&PostRAHazardRecognizerID);
 
+  addPass(&AMDGPUWaitSGPRHazardsLegacyID);
```

Comment:
How does this interact with InsertDelayAlu? Is this intentionally right before it?

Reply:
In theory InsertDelayAlu can adjust `s_delay_alu` settings based on presence of `s_wait_alu` instructions, but I don't think support for this tracking yet exists.
Pass position was to place it after PostRAHazardRecognizer from where functionality was moved.

Reply:
> In theory InsertDelayAlu can adjust s_delay_alu settings based on presence of s_wait_alu instructions

Yes `AMDGPUInsertDelayAlu` does account for this, in that it knows that an extra `s_wait_alu` instruction will cause an extra single cycle issue delay which can avoid the need for an `s_delay_alu`. This does happen in some cases; see inline comment.